### DISCUSSION TITLE
Can anyone here confirm this ADC board will work?

### DISCUSSION THREAD

**Discussion by john-** [pschatzmann/arduino-audio-tools#430] [Q&A] [answered]
I cannot find the exact ADC board referenced in the Arduino Audio Tools wiki:

https://github.com/pschatzmann/arduino-audio-tools/wiki/External-ADC#i2s-adc-audio-i2s-capture-card-module

I find one that is close but not the same:

https://www.aliexpress.us/item/2251832803602142.html

Differences based on visual inspection:

- Mute pin is replaced by a ground pin (Audio ground)?
- Jumpers have a different purpose (master/slave and 16/24bit)

Anyone here have experience with this ADC as to whether it will work with AAT?  Also, I would be find with a different ADC board if it was known to work.

**Comment by TangerineDreamer** [comment, marked as answer]
Hi John,
Got this one exactly the same as yours, i was a little bit confused,but had good results after a while,here is the sketch with
wiring configuring,i use as output PC5102A 👍 :

//----------------------------------------------------------------------------------------


/*I2S OUT: DAC MODULE PC5102A
  VIN->5V,GND->GND,SCK->GND,BCK->18,
  DIN->23,WSEL (WS) = LRCK = LCK->19,
 * 
*I2Sin IN : ADC MODULE: SLAVE,16BITS
 JUMPER ON : MCLK-->24,576M
 VIN->5V,GND->GND,SCK->GND,BCK->14,
 DIN->22,LRCK->15,
 */

#include "AudioTools.h"
 
uint16_t sample_rate=32000;
uint16_t channels = 2;
uint16_t bits_per_sample = 16; // or try with 24 or 32

I2SStream in;
I2SStream out; 
StreamCopy copier(out, in); // copies sound into i2s


// Arduino Setup
void setup(void) {  
  // Open Serial 
  Serial.begin(115200);
  // change to Warning to improve the quality
  AudioLogger::instance().begin(Serial, AudioLogger::Info); 

  // start I2S in
  Serial.println("starting I2S...");
  auto config_in = in.defaultConfig(RX_MODE);
  config_in.sample_rate = sample_rate; 
  config_in.bits_per_sample = bits_per_sample; 
  config_in.i2s_format = I2S_MSB_FORMAT;
  config_in.is_master = true;
  config_in.port_no = 0;
  config_in.pin_bck = 14;
  config_in.pin_ws = 15;
  config_in.pin_data = 22;
  // config_in.fixed_mclk = sample_rate * 256
  // config_in.pin_mck = 2
  in.begin(config_in);

  // start I2S out
  auto config_out = out.defaultConfig(TX_MODE);
  config_out.sample_rate = sample_rate; 
  config_out.bits_per_sample = bits_per_sample;
  config_out.i2s_format = I2S_MSB_FORMAT;
  config_out.is_master = true;
  config_out.port_no = 1;
  config_out.pin_bck = 18;
  config_out.pin_ws = 19;
  config_out.pin_data = 23;
  out.begin(config_out);

  Serial.println("I2S started...");
}

// Arduino loop - copy sound to out 
void loop() {
  copier.copy();
}

//---------------------------------------------------------------------------------------------

I had the best results with 32000 sample rate,but it was 2 months ago,maybe the library had some tweaking since :)
Are you talking about AGND for muting?
i really didn't know what it was really about.

**Reply by john-** [reply]
Thanks for the quick and detailed response!

In my application I don't think I will need to mute on the ADC side.  I was just commenting on that difference between the image in the Aurduino Audio Tools wiki  and what I found on AliExpress.

**Comment by TangerineDreamer** [comment]
If you success to make it work with 44100 sample rate,please let me know,but i believe you have not bought it yet,
it's very hard to find a input board not so expensive that works with the shatzmann library,vs1053 is good but you
can't get the i2s to feed the input or output of esp32 for  bluetooth a2p streaming,audiokit seems the all in one
solution.

**Reply by john-** [reply]
I haven't bought it yet but will give 44.1K sample rate a try when I get one.

I am not quite understanding your comment on Bluetooth A2DP streaming.  My plan was to have the ADC connected to the ESP32 via I2S.  When there was no A2DP connection it would send the ADC I2S data out over I2S to a DAC.  If there is an A2DP connection it would send that audio data to the same DAC (via I2S).

Is that related to your comment?

**Comment by TangerineDreamer** [comment]
It's a i2s>i2s connection,the adc > i2s(input) into the esp32 and the esp32 make the i2s(stream) go to i2s-out(stream)
or if you want to transmit by bluetooth,go to A2DP(stream),the library makes this connections inside,there is no physical
i2s connection,only logicial.

**Reply by john-** [reply]
Sounds good thanks again.
